pyuvdata/uvbeam.py

```diff
@@ -978,6 +977,43 @@ def __iadd__(self, other):
         self.__add__(other, inplace=True)
         return self
 
+    def get_beam_area(self,stokes='I'):
+        """
+        Computes the integral of the beam, which has units of steradians
+        """
+        if self.beam_type is not 'power':
+            raise ValueError('beam_type must be power')
+        if self._data_normalization.value != 'peak':
+            print self._data_normalization.value
+            raise ValueError('beam must be peak normalized')
+        npix = self.data_array.shape[-1]
+        if stokes == 'I':
+            # Assume A_I = (B_xx + B_yy)/2
+            beam = np.mean(self.data_array[0,0], axis=-3)
+        else:
```

Comment:
This worries me for several reasons.

First, the first index is the vector index. Usually it only has length one for power beams, but that's not required. If it is a power beam with Naxes_vec > 1, then it's reporting the power separately in different directions. I suspect that you either want to error on that or sum over that axis. (You can get such a beam for testing by reading in an efield beam and then converting to power using `efield_to_power` and setting `keep_basis_vector=True`).

Even more critically, you need to be careful about the polarization axis. The polarization axis can be in stokes parameters already (values in the pol array between 1-4), or it can be any set of linear pols (xx, yy, xy, yx -> -5:-8) _in any order_. Do you really want to sum over all 4 if they are all present? You can test with a beam with all 4 pols by reading in an efield beam and converting to power with `efield_to_power` -- by default it also calculates the xy & yx power beams.

Finally, I'm not sure I like calling this options 'stokes=I' because summing over the xx & yy polarizations in this fashion is really not what one means by a stokes I beam in general (although I'd need to check details with James, I believe that a stokes I beam has contributions from all 4 pols and they need to be combined in different ways in different parts of the sky -- in fact the conversion should use a different 4x4 matrix for every sky pixel). I'd prefer something that's less overloaded.

Reply:
Definitely! I've changed a bunch of things:
---If Naxes_vec > 1, the code will now error.
---The code now checks to see if a Stokes I beam is available. If it is, it'll use that. Otherwise, it will extract only xx and yy and average those.
---Agreed that it's not really Stokes I. More like what we've sometimes called "pseudo Stokes I". The terminology has changed now, and I've also added a journal reference.